apps/ui/src/views/account/AccountProfileView.vue

```diff
@@ -0,0 +1,216 @@
+                  editableValues.graduation.dialog = false;
+                }">SAVE</v-btn>
+              </template>
+            </v-card>
+          </template>
+        </v-dialog>
+      </div>
+    </div>
+    <v-snackbar :color="snackbarColor" v-model="snackbar" :timeout="3000">
+      {{ snackbarText}}
+    </v-snackbar>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { useAuthStore } from "@/stores/auth";
+import { ref, onMounted } from "vue";
+import { useQuery, useMutation } from "@vue/apollo-composable";
+import { GenerateOwnPersonDocument, UpdatePersonDocument, UserDetailsDocument } from "@/graphql/types";
+import type { Person, User } from "@/graphql/types";
+import moment from "moment";
+
+
+const auth = useAuthStore();
+const generatePerson = useMutation(GenerateOwnPersonDocument);
+const mutatePerson = useMutation(UpdatePersonDocument);
+
+const userId = ref<number|null>(await auth.getOwnId());
+const userData = ref<Partial<User>>({});
+const personData = ref<Partial<Person>>({});
+const profileCreation = ref();
+const profileName = ref<string>("");
+const editableValues = ref({
+  name: {
+    dialog: false,
+    value: ""
+  },
+  pronouns: {
+    dialog: false,
+    value: ""
+  },
+  graduation: {
+    dialog: false,
+    value: ""
+  },
+});
+const pronouns = ["he/him", "she/her", "they/them"];
+const nameRule = (v: string) => {
+  if (!(v.trim()))
+    return "Name is required";
+  return true;
+};
+const snackbar = ref(false);
+const snackbarColor = ref("green");
+const snackbarText = ref("");
+
+async function createProfile() {
+  await profileCreation.value.validate();
+  if (profileName.value.trim()) {
+    try {
+      await generatePerson.mutate({
+        name: profileName.value
+      });
+      snackbar.value = true;
+      snackbarColor.value = "green";
+      snackbarText.value = "Success"
+    } catch (e) {
+      console.error(e);
+      snackbar.value = true;
+      snackbarColor.value = "red";
+      snackbarText.value = "An error has occurred";
+    }
+    userQuery.refetch();
+  }
+}
+const userQuery = useQuery(UserDetailsDocument, {
+  id: userId.value
+});
+
+userQuery.onResult((result) => {
+  if (result.data) {
+    userData.value = structuredClone(result.data.user) as User;
+    if (result.data.user?.person) {
+      personData.value = structuredClone(result.data.user.person);
+      personData.value.graduation = new Date(personData.value.graduation);
+      editableValues.value.name.value = personData.value.name as string;
+      editableValues.value.pronouns.value = personData.value.pronouns as string;
+      editableValues.value.graduation.value = personData.value.graduation;
+    }
+  }
+});
+
+async function saveChanges(text: string) {
+  await mutatePerson.mutate({
+    id: personData.value.id,
+    data: {
+      name: personData.value.name,
+      pronouns: personData.value.pronouns,
+      graduation: personData.value.graduation
```

Comment:
If graduation is null, always sets to 01-01-1970, regardless of whether it was actually touched/modified in the UI at all